core/cthreads.c

```diff
@@ -0,0 +1,246 @@
+                if (attr->prioceiling) pthread_mutexattr_setprioceiling(&pAttr, attr->prioceiling);
+            #endif
+        }
+
+        return pthread_mutex_init(&mutex->pMutex, attr ? &pAttr : NULL);
+    #endif
+}
+
+int cthreads_mutex_lock(struct cthreads_mutex *mutex) {
+    #ifdef _WIN32
+        return WaitForSingleObject(mutex->wMutex, INFINITE) == WAIT_OBJECT_0 ? 0 : 1; // Needs to re-see
+    #else
+        return pthread_mutex_lock(&mutex->pMutex);
+    #endif
+}
+
+int cthreads_mutex_trylock(struct cthreads_mutex *mutex) {
+    #ifdef _WIN32
+        return WaitForSingleObject(mutex->wMutex, 0) == WAIT_OBJECT_0 ? 0 : 1; // Needs to re-see
+    #else
+        return pthread_mutex_trylock(&mutex->pMutex);
+    #endif
+}
+
+int cthreads_mutex_unlock(struct cthreads_mutex *mutex) {
+    #ifdef _WIN32
+        return ReleaseMutex(mutex->wMutex) == 0 ? 1 : 0; // Needs to re-see
+    #else
+        return pthread_mutex_unlock(&mutex->pMutex);
+    #endif
+}
+
+int cthreads_mutex_destroy(struct cthreads_mutex *mutex) {
+    #ifdef _WIN32
+        return CloseHandle(mutex->wMutex) == 0 ? 1 : 0; // Needs to re-see
+    #else
+        return pthread_mutex_destroy(&mutex->pMutex);
+    #endif
+}
+
+int cthreads_cond_init(struct cthreads_cond *cond, struct cthreads_cond_attr *attr) {
+    #ifdef _WIN32
+        if (attr) cond->wCond = CreateEvent(NULL, attr->bManualReset ? TRUE : FALSE, 
+                                            attr->bInitialState ? TRUE : FALSE,
+                                            attr->lpName ? (LPTSTR)attr->lpName : NULL);
+        else cond->wCond = CreateEvent(NULL, FALSE, FALSE, NULL);
+
+        if (cond->wCond == NULL) return 1;
+        return 0;
+    #else
+        pthread_condattr_t pAttr;
+        if (attr) {
+            if (pthread_condattr_init(&pAttr) != 0) return 1;
+            if (attr->pshared) pthread_condattr_setpshared(&pAttr, attr->pshared);
+            if (attr->clock) pthread_condattr_setclock(&pAttr, attr->clock);
+        }
+
+        return pthread_cond_init(&cond->pCond, attr ? &pAttr : NULL);
+    #endif
+}
+
+int cthreads_cond_signal(struct cthreads_cond *cond) {
+    #ifdef _WIN32
+        return SetEvent(cond->wCond) == 0 ? 1 : 0; // Needs to re-see
+    #else
+        return pthread_cond_signal(&cond->pCond);
+    #endif
+}
+
+int cthreads_cond_broadcast(struct cthreads_cond *cond) {
+    #ifdef _WIN32
+        return SetEvent(cond->wCond) == 0 ? 1 : 0; // Needs to re-see
+    #else
+        return pthread_cond_broadcast(&cond->pCond);
+    #endif
+}
+
+int cthreads_cond_destroy(struct cthreads_cond *cond) {
+    #ifdef _WIN32
+        return CloseHandle(cond->wCond) == 0 ? 1 : 0; // Needs to re-see
+    #else
+        return pthread_cond_destroy(&cond->pCond);
+    #endif
+}
+
+int cthreads_cond_wait(struct cthreads_cond *cond, struct cthreads_mutex *mutex) {
+    #ifdef _WIN32
+        return SleepConditionVariableCS(&cond->wCond, &mutex->wMutex, INFINITE) == 0 ? 1 : 0;
```

Comment:
TODO for me: check to see that this works on pre-Windows 2000 systems just for fun

Reply:
It would be wonderful if you could test this function in many Windows compilers, those functions which doesn't have a direct replacement in Windows threads were insanely hard to find, and may be not totally correct